### PR TITLE
travis: use prebuilt image

### PR DESCRIPTION
**Pull request by liushuyu**
This is a WIP change to the current CI workflow.

In this PR, the builder no longer install packages during the building process, instead, the prebuilt Docker images from https://hub.docker.com/r/citraemu/build-environments/ are used.

This can ensure the consistency across the builds and frozen builds are absolutely frozen to the exact version required.

There are several problems in this PR. They will be ironed out very soon. In the mean time, you can stlll comment about your concerns against the change.

MinGW builds will have seperate commits to address the changes. (b/c it's unsure if we will use cross-compile to produce Windows (R) binaries)

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3839)

### CODE REVIEW

**Comment by liushuyu** [comment]
Tagging @jroweboy and @j-selby

File: .travis/linux-frozen/build.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash -ex
 mkdir -p "$HOME/.ccache"
-docker pull ubuntu:18.04
-docker run --env-file .travis/common/travis-ci.env -v $(pwd):/citra -v "$HOME/.ccache":/root/.ccache ubuntu:18.04 /bin/bash -ex /citra/.travis/linux-frozen/docker.sh
+docker pull citraemu/build-environments:linux-frozen
+docker run --env-file .travis/common/travis-ci.env -v $(pwd):/citra -v "$HOME/.ccache":/root/.ccache citraemu/build-environments:linux-fresh /bin/bash -ex /citra/.travis/linux-frozen/docker.sh
```
Comment (j-selby) [inline .travis/linux-frozen/build.sh]: Pulling frozen image, running fresh?

Comment (j-selby) [reply .travis/linux-frozen/build.sh]: Would also be nice to see a `linux-frozen/deps.sh` equivalent as well to split up pulling/running, though don't feel obligated to fix that one.

Comment (liushuyu) [reply .travis/linux-frozen/build.sh]: I'll address this later.

Comment (liushuyu) [reply .travis/linux-frozen/build.sh]: I'll address this later.

**Comment by liushuyu** [comment]
Rebased with new commits from `master` and added MinGW changes

**Comment by jroweboy** [comment]
> There are several problems in this PR. They will be ironed out very soon. In the mean time, you can stlll comment about your concerns against the change.

Is this still the case, or am I fine to tag this for canary now?

**Comment by liushuyu** [comment]
>Is this still the case, or am I fine to tag this for canary now?

They should have mostly resolved now. You can tag it `canary` at current stage.

**Comment by wwylele** [comment]
Sorry for not following this PR... what is the status right now? Is it ready for merge into master or merge into canary?

**Comment by liushuyu** [comment]
>Sorry for not following this PR... what is the status right now? Is it ready for merge into master or merge into canary?

Better goes into `canary` first.

**Comment by jroweboy** [comment]
Planning on merging this in two days from now if we don't find any issues. Thanks for the contribution!

**Pull request merged**